indexing-solr/src/main/java/edu/unc/lib/boxc/indexing/solr/filter/SetDescriptiveMetadataFilter.java

```diff
@@ -264,6 +265,30 @@ private void extractSubjects(Element mods, IndexDocumentBean idb) {
 
     }
 
+    private void extractLocations(Element mods, IndexDocumentBean idb) {
+        List<Element> locationEls = mods.getChildren("subject", JDOMNamespaceUtil.MODS_V3_NS);
+        List<String> locations = new ArrayList<>();
+        if (locationEls.size() > 0) {
+            for (Element locationEl : locationEls) {
+                String possibleLocation = locationEl.getAttributeValue("authority");
+                if (possibleLocation != null && possibleLocation.equals("lcsh")) {
+                    List<Element> locationParts = locationEl.getChildren();
```

Comment:
you can directly list children with a particular name if you're only interested in one type of tag
```
locationEl.getChildren("geographic", JDOMNamespaceUtil.MODS_V3_NS);
```